coordinator/performance/monitor.go

```diff
@@ -0,0 +1,79 @@
+package performance
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/mattermost/mattermost-load-test-ng/coordinator/performance/prometheus"
+
+	"github.com/mattermost/mattermost-server/v5/mlog"
+)
+
+type Monitor struct {
+	config     MonitorConfig
+	helper     *prometheus.Helper
+	stopChan   chan struct{}
+	statusChan chan Status
+}
+
+func NewMonitor(config MonitorConfig) (*Monitor, error) {
+	if ok, err := config.IsValid(); !ok {
+		return nil, err
+	}
+	helper, err := prometheus.NewHelper(config.PrometheusURL)
+	if err != nil {
+		return nil, fmt.Errorf("performance: failed to create prometheus.Helper: %w", err)
+	}
+	return &Monitor{
+		config:     config,
+		helper:     helper,
+		stopChan:   make(chan struct{}),
+		statusChan: make(chan Status),
+	}, nil
+}
+
+func (m *Monitor) Run() (<-chan Status, error) {
+	go func() {
+		mlog.Info("monitor: started")
+		for {
+			m.statusChan <- m.runQueries()
+			select {
+			case <-m.stopChan:
+				mlog.Info("monitor: shutting down")
+				return
+			case <-time.After(time.Duration(m.config.UpdateInterval) * time.Millisecond):
+			}
+		}
+	}()
+	return m.statusChan, nil
+}
+
+func (m *Monitor) Stop() {
+	close(m.stopChan)
```

Comment:
We should have a way to pass this signal into the runQueries loop. It seems likely that there may be 20-30 queries and the loop is stuck in runQueries and the stop signal never reaches until it exits from runQueries.

We can have a context variable to cancel the loop or pass the signal into the loop. I am fine with whatever you think is better.